runner/utils/driver/istio/helpers.go

```diff
@@ -252,10 +252,51 @@ func prepareVirtualServiceHTTPRoutes(ctx context.Context, projectID, serviceID s
 			}
 		}
 
+		matchers := make([]*networkingv1alpha3.HTTPMatchRequest, 0)
+		for _, matcher := range route.Matchers {
+			tempMatcher := new(networkingv1alpha3.HTTPMatchRequest)
+
+			// Add url matchers
+			if matcher.URL != nil {
+				tempMatcher.IgnoreUriCase = matcher.URL.IgnoreCase
+				tempMatcher.Uri = new(networkingv1alpha3.StringMatch)
+				switch matcher.URL.Type {
+				case model.RouteHTTPMatchTypeExact:
+					tempMatcher.Uri.MatchType = &networkingv1alpha3.StringMatch_Exact{Exact: matcher.URL.Value}
+				case model.RouteHTTPMatchTypePrefix:
+					tempMatcher.Uri.MatchType = &networkingv1alpha3.StringMatch_Prefix{Prefix: matcher.URL.Value}
+				case model.RouteHTTPMatchTypeRegex:
+					tempMatcher.Uri.MatchType = &networkingv1alpha3.StringMatch_Regex{Regex: matcher.URL.Value}
+				}
+			}
+
+			// 	Add header matchers
+			tempMatcher.Headers = map[string]*networkingv1alpha3.StringMatch{}
+			for _, header := range matcher.Headers {
+				switch header.Type {
+				case model.RouteHTTPMatchTypeExact:
+					tempMatcher.Headers[header.Key] = &networkingv1alpha3.StringMatch{MatchType: &networkingv1alpha3.StringMatch_Exact{Exact: header.Value}}
+				case model.RouteHTTPMatchTypePrefix:
+					tempMatcher.Headers[header.Key] = &networkingv1alpha3.StringMatch{MatchType: &networkingv1alpha3.StringMatch_Prefix{Prefix: header.Value}}
+				case model.RouteHTTPMatchTypeRegex:
+					tempMatcher.Headers[header.Key] = &networkingv1alpha3.StringMatch{MatchType: &networkingv1alpha3.StringMatch_Regex{Regex: header.Value}}
+				case model.RouteHTTPMatchTypeCheckPresence:
+					tempMatcher.Headers[header.Key] = &networkingv1alpha3.StringMatch{}
```

Comment:
Are you sure it doesn't have to be nil? I think we should test it manually once

Reply:
We cannot assign nil if we try to assign nil then istio throws error.
I have tested this & it works